src/super_gradients/training/datasets/detection_datasets/pascal_voc_format_detection.py

```diff
@@ -0,0 +1,141 @@
+
+from super_gradients.common.deprecate import deprecated_parameter
+from super_gradients.common.registry.registry import register_dataset
+from super_gradients.training.utils.utils import get_image_size_from_path
+from super_gradients.training.datasets.detection_datasets.detection_dataset import DetectionDataset
+from super_gradients.training.datasets.data_formats.default_formats import XYXY_LABEL
+from super_gradients.common.abstractions.abstract_logger import get_logger
+
+logger = get_logger(__name__)
+
+
+@register_dataset("PascalVOCFormatDetectionDataset")
+class PascalVOCFormatDetectionDataset(DetectionDataset):
+    """Dataset for Pascal VOC object detection
+
+        Parameters:
+            data_dir (str): Base directory where the dataset is stored.
+            images_dir (str): Directory containing all the images, relative to `data_dir`. Defaults to None.
+            labels_dir (str): Directory containing all the labels, relative to `data_dir`. Defaults to None.
+
+        Dataset structure:
+
+        ./data/pascal_voc
+        ├─images
+        │   ├─ train2012
+        │   ├─ val2012
+        │   ├─ VOCdevkit
+        │   │    ├─ VOC2007
+        │   │    │  ├──JPEGImages
+        │   │    │  ├──SegmentationClass
+        │   │    │  ├──ImageSets
+        │   │    │  ├──ImageSets/Segmentation
+        │   │    │  ├──ImageSets/Main
+        │   │    │  ├──ImageSets/Layout
+        │   │    │  ├──Annotations
+        │   │    │  └──SegmentationObject
+        │   │    └──VOC2012
+        │   │       ├──JPEGImages
+        │   │       ├──SegmentationClass
+        │   │       ├──ImageSets
+        │   │       ├──ImageSets/Segmentation
+        │   │       ├──ImageSets/Main
+        │   │       ├──ImageSets/Action
+        │   │       ├──ImageSets/Layout
+        │   │       ├──Annotations
+        │   │       └──SegmentationObject
+        │   ├─train2007
+        │   ├─test2007
+        │   └─val2007
+        └─labels
+            ├─train2012
+            ├─val2012
+            ├─train2007
+            ├─test2007
+            └─val2007
+
+    Note:
+        If both 'images_sub_directory' and ('images_dir', 'labels_dir') are provided, a warning will be raised.
+
+    Usage:
+        voc_2012_train = PascalVOCDetectionDataset(data_dir="./data/pascal_voc",
+                                            images_dir="images/train2012/JPEGImages",
+                                            labels_dir="labels/train2012/Annotations",
+                                            download=True)
+    """
+
+    @deprecated_parameter(
+        "images_sub_directory",
+        deprecated_since="3.7.0",
+        removed_from="3.8.0",
+        reason="Support of `images_sub_directory` is removed since it allows less flexibility." " Please use 'images_dir' and 'labels_dir' instead.",
+    )
+    def __init__(
+        self,
+        images_dir: Optional[str],
+        labels_dir: Optional[str],
+        *args,
+        **kwargs,
```

Comment:
A `*args, **kwargs` is a chaotic evil move. Can we have explicit and full list of arguments we can use in this class's c'tor?

Reply:
done